v3/util/fqdn.go

```diff
@@ -117,3 +118,13 @@ func CommonNameIsIP(cert *x509.Certificate) bool {
 		return true
 	}
 }
+
+var nonLDHCharacterRegex = regexp.MustCompile(`[^a-zA-Z0-9\-]`)
+
+func IsLDHLabel(label string) bool {
+	return len(label) > 0 &&
+		len(label) <= 63 &&
+		!nonLDHCharacterRegex.MatchString(label) &&
+		!strings.HasPrefix(label, "-") &&
+		!strings.HasSuffix(label, "-")
```

Comment:
```suggestion
		!strings.HasSuffix(label, "-") &&
		!(util.HasReservedLabelPrefix(label) && !util.HasXNLabelPrefix(label))
```

From https://github.com/zmap/zlint/blob/cb17369b4628c684ac68c1fc169ff2a38c00cfdf/v3/lints/cabf_br/lint_dnsname_contains_prohibited_reserved_label.go#L51